rcl_action/src/rcl_action/action_server.c

```diff
@@ -590,6 +685,20 @@ rcl_action_expire_goals(
   return ret_final;
 }
 
+rcl_ret_t
+rcl_action_notify_goal_done(
+  const rcl_action_server_t * action_server,
+  rcl_action_goal_handle_t * goal_handle)
+{
+  (void)goal_handle;
```

Comment:
Why is this unused here? Are you expecting it to be expired and remove from the `goal_handles` automatically in `_recalculate_expire_timer()`? If so would it make sense to use `goal_handle` to ensure that did happen before returning?

Reply:
I don't have a good reason for passing it in. I'll remove the `goal_handle` as a parameter.

 It felt like the action server should have the goal handle in case it wanted to do something with it in the future, but the only thing that needs to happen when a goal reaches a terminal state is to check if the expire timer period needs to be set.

Reply:
Removed parameter in 2aea0e3db0be5fcb002e5d61f2cc5efb747a44a1